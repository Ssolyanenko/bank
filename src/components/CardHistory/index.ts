export * from './History';
export * from './Pagination';
export * from './CardHistory';
