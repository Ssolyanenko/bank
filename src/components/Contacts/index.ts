export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
