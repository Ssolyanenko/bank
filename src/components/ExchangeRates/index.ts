export * from './ExchangeRates';
export * from './Converter';
export * from './Rates';
