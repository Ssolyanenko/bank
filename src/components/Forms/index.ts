export * from './Login';
export * from './PasswordRecoveryForm';
export * from './GeneralInfo';
export * from './Security';
export * from './OrderDebitTravelCard';
export * from './SetPinCodeForm';
export * from './OrderCreditCardPremium';
export * from './FormNotifications';
export * from './FilterHistoryForm';
export * from './CardLimitForm';
export * from './OrderCreditCardBilliable';
export * from './OrderDebitSmartCard';
export * from './TransferByPhoneNumberForm';
export * from './AcceptTerms';
