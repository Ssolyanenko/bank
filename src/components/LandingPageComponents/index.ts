export * from './LandingPageHeader';
export * from './LandingPageMain';
export * from './LandingPageFooter';
