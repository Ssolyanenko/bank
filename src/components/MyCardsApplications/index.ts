export * from './MyCardApplications';
export * from './ApplicationStatus';
