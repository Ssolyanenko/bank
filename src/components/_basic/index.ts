export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
