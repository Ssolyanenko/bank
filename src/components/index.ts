export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
