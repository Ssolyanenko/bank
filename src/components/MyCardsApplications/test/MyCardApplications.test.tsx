import { shallow } from 'enzyme';

import { CARD_APPLICATIONS } from 'constants/cardApplications';
import { MyCardApplications } from 'components';

describe('<MyCardApplications />', (): void => {
  test('should make a snapshot with Card Applications', (): void => {
    const wrapper = shallow(<MyCardApplications cardApplicationsList={CARD_APPLICATIONS} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should make a snapshot without Card Applications', (): void => {
    const wrapper = shallow(<MyCardApplications cardApplicationsList={[]} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });
});
