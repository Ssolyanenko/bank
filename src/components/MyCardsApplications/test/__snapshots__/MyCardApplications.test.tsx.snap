// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyCardApplications /> should make a snapshot with Card Applications 1`] = `
<div
  className="wrapper"
>
  <h2
    className="pageTitle"
  >
    My card applications
  </h2>
  <ul
    className="cardApplicationsList"
  >
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Express delivery by courier
        </span>
        <span
          className="applicationCardType"
        >
          Smart Debit card
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          02.08.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={1}
        status="Finalized"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Parcel by mail
        </span>
        <span
          className="applicationCardType"
        >
          Debit Travel card
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          12.10.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={2}
        status="Approved"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Express delivery by courier
        </span>
        <span
          className="applicationCardType"
        >
          Smart Debit card
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          09.11.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={3}
        status="On the way"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Pick up at the bank
        </span>
        <span
          className="applicationCardType"
        >
          Credit Card Billable Premium
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          05.09.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={4}
        status="Rejected"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Parcel by mail
        </span>
        <span
          className="applicationCardType"
        >
          Credit Card Billable
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          10.07.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={5}
        status="Ready for pick up"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Pick up at the bank
        </span>
        <span
          className="applicationCardType"
        >
          Credit Card Billable Premium
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          05.09.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={6}
        status="Under consideration"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Parcel by mail
        </span>
        <span
          className="applicationCardType"
        >
          Credit Card Billable Premium
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          05.09.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={7}
        status="Ready for pick up"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Express delivery by courier
        </span>
        <span
          className="applicationCardType"
        >
          Smart Debit card
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          09.11.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={8}
        status="Collected"
      />
    </li>
    <li
      className="cardApplication"
    >
      <div
        className="applicationInfo fixedWidth"
      >
        <span
          className="applicationMethod"
        >
          Pick up at the bank
        </span>
        <span
          className="applicationCardType"
        >
          Credit Card Billable Premium
        </span>
      </div>
      <div
        className="applicationInfo textAlignCenter"
      >
        <span
          className="applicationMethod"
        >
          Submission date
          :
        </span>
        <span
          className="applicationCardType"
        >
          05.09.2022
        </span>
      </div>
      <ApplicationStatus
        cardId={9}
        status="Under consideration"
      />
    </li>
  </ul>
</div>
`;

exports[`<MyCardApplications /> should make a snapshot without Card Applications 1`] = `
<div
  className="wrapper"
>
  <h2
    className="pageTitle"
  >
    My card applications
  </h2>
  <InformationBlock>
    You don't have any cards applications. You may check our Card Products and order a card online
  </InformationBlock>
</div>
`;
